prober/handler.go

```diff
@@ -30,7 +30,7 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/prometheus/common/expfmt"
 	"github.com/prometheus/common/promslog"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
```

Comment:
I see @SuperQ's comment about blockers around upgrading to v3. https://github.com/go-yaml/yaml/issues/665#issuecomment-1145736596

so I will hold and wait for @SuperQ's review on this before we merge this.